ocs_ci/ocs/resources/pod.py

```diff
@@ -143,6 +145,59 @@ def exec_ceph_cmd(self, ceph_cmd, format='json-pretty'):
             return [item for item in out if item]
         return out
 
+    def run_io(
+        self, storage_type, size, io_direction='rw', rw_ratio=75,
+        jobs=1, runtime=60,
+    ):
+        """
+        Execute FIO on a pod
+        This operation will run in background and will store the results in
+        'self.thread.result()'.
+        In order to wait for the output and not continue with the test until
+        FIO is done, call self.thread.result() right after calling run_io.
+        See tests/manage/test_pvc_deletion_during_io.py::test_run_io
+        for usage of FIO
+
+        Args:
+            storage_type (str): 'fs' or 'block'
+            size (str): Size in MB, e.g. '200M'
+            io_direction (str): Determines the operation:
+                'ro', 'wo', 'rw' (default: 'rw')
+            rw_ratio (int): Determines the reads and writes using a
+                <rw_ratio>%/100-<rw_ratio>%
+                (e.g. the default is 75 which means it is 75%/25% which
+                equivalent to 3 reads are performed for every 1 write)
+            jobs (int): Number of jobs to execute FIO
+            runtime (int): Number of seconds IO should run for
+        """
+        name = 'test_workload'
+        spec = self.pod_data.get('spec')
+        path = (
+            spec.get('containers')[0].get('volumeMounts')[0].get(
+                'mountPath'
+            )
+        )
+        work_load = 'fio'
+        # few io parameters for Fio
+
+        wl = workload.WorkLoad(
+            name, path, work_load, storage_type, self, jobs
+        )
+        assert wl.setup(), "Setup up for FIO failed"
+        if io_direction == 'rw':
+            io_params = templating.load_yaml_to_dict(
+                constants.FIO_IO_RW_PARAMS_YAML
+            )
+            io_params['rwmixread'] = rw_ratio
+        else:
+            io_params = templating.load_yaml_to_dict(
+                constants.FIO_IO_PARAMS_YAML
+            )
+        io_params['runtime'] = runtime
+        io_params['size'] = size
+
+        self.thread = wl.run(**io_params)
```

Comment:
I feel self.thread is misnomer here, because a concurrent.future object provides many methods like result(), done() etc, if user assumes that this is pure python thread object then he may try thread operations on this object.

Reply:
done